include/InverseKinematicsIPOPT.h

```diff
@@ -8,94 +8,92 @@
 #include <iDynTree/Core/Transform.h>
 #include <iDynTree/Model/Model.h>
 #include <iDynTree/ModelIO/ModelLoader.h>
-#include <iDynTree/Core/EigenHelpers.h>
 #include <iDynTree/KinDynComputations.h>
-#include <math.h>
 #include <IpIpoptApplication.hpp>
-#include <string>
-#include <vector>
 #include <Eigen/Core>
-#include <iDynTree/Core/EigenHelpers.h>
+
 
 class InverseKinematicsIPOPT : public Ipopt::TNLP {
     iDynTree::Model model;
     bool modelLoaded;
     bool gainsLoaded;
-    bool framesLoaded;
     iDynTree::KinDynComputations iKDC;
     iDynTree::MatrixDynSize hessian;
     iDynTree::VectorDynSize gradient;
     iDynTree::FrameIndex parentFrame, endEffectorFrame;
-    std::vector< std::pair<double,double> > jointsLimits; //joints upper and lower limits 
+    std::vector< std::pair< double,double > > jointsLimits; //joints upper and lower limits
     int totalDOF;
-    iDynTree::MatrixDynSize Ep, Eq, Edof; //extractors from the total variable vector    
+    iDynTree::MatrixDynSize Ep, Eq, Edof; //extractors from the total variable vector
+    iDynTree::FrameFreeFloatingJacobian e_J_we, p_J_wp;
+    iDynTree::MatrixDynSize jacobian;
+    iDynTree::VectorDynSize jointsTemp;
+    Eigen::Matrix< double, Eigen::Dynamic, Eigen::Dynamic, Eigen::RowMajor > denseJac;
+    iDynTree::Vector3 positionResult;
+    iDynTree::Vector4 quaternionResult;
+    
+    
+    void removeJoints(const iDynTree::Model modelInput);
+    
+    void twistToQuaternionTwist(iDynTree::Vector4 &quaternion, iDynTree::MatrixFixSize<7,6>& mapOut);
+
+    void relativeJacobian(const iDynTree::VectorDynSize &configuration, iDynTree::MatrixDynSize& jacobianOut);
     
 public:
-    iDynTree::VectorDynSize jointMap;
     iDynTree::VectorDynSize jointResult;
     iDynTree::Vector3 gains;
     iDynTree::Vector3 desiredPosition;
     iDynTree::Vector4 desiredQuaternion; 
     iDynTree::VectorDynSize desiredJoints;
-    iDynTree::Vector3 positionError;
-    iDynTree::Rotation rotationError;
-    double angleError;
     signed int exitCode;
     
     InverseKinematicsIPOPT();
     
-    virtual ~InverseKinematicsIPOPT();
+    ~InverseKinematicsIPOPT();
```

Comment:
So in this case I guess yes :grin:. Does it hold for all the inherited methods? What happens if I don't set it as ``virtual``?